torchgeo/datasets/patternnet.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""PatternNet dataset."""
+
+import os
+from typing import Callable, Dict, Optional
+
+import torch
+from torch import Tensor
+from torchvision.datasets import ImageFolder
+from torchvision.transforms import PILToTensor
+
+from .geo import VisionDataset
+from .utils import download_and_extract_archive
+
+
+class PatternNet(VisionDataset, ImageFolder):  # type: ignore[misc]
+    """PatternNet dataset.
+
+    The `PatternNet <https://sites.google.com/view/zhouwx/dataset>`_
+    dataset is a dataset for remote sensing scene classification and image retrieval.
+
+    Dataset features:
+    * 30,400 images with 6-50 cm per pixel resolution (256x256 px)
+    * three spectral bands - RGB
+    * 38 scene classes, 800 images per class
+
+    Dataset format:
+    * images are three-channel jpgs
+
+    Dataset classes:
+    0. airplane
+    1. baseball_field
+    2. basketball_court
+    3. beach
+    4. bridge
+    5. cemetery
+    6. chaparral
+    7. christmas_tree_farm
+    8. closed_road
+    9. coastal_mansion
+    10. crosswalk
+    11. dense_residential
+    12. ferry_terminal
+    13. football_field
+    14. forest
+    15. freeway
+    16. golf_course
+    17. harbor
+    18. intersection
+    19. mobile_home_park
+    20. nursing_home
+    21. oil_gas_field
+    22. oil_well
+    23. overpass
+    24. parking_lot
+    25. parking_space
+    26. railway
+    27. river
+    28. runway
+    29. runway_marking
+    30. shipping_yard
+    31. solar_panel
+    32. sparse_residential
+    33. storage_tank
+    34. swimming_pool
+    35. tennis_court
+    36. transformer_station
+    37. wastewater_treatment_plant
+
+    If you use this dataset in your research, please cite the following paper:
+    * https://www.sciencedirect.com/science/article/abs/pii/S0924271618300042
```

Comment:
```suggestion
    * https://doi.org/10.1016/j.isprsjprs.2018.01.004
```
I prefer to use the DOI when possible, it's shorter

Reply:
Updated